src/tester.rs

```diff
@@ -385,4 +397,57 @@ mod test {
         let expected_argument = format!("{:?}", [true, true]);
         assert_eq!(failing_case.arguments, vec![expected_argument]);
     }
+
+    #[test]
+    fn floats_finite_only_flag_is_set_then_all_the_generated_floats_are_finite() {
+        let floats_finite_only = true;
+
+        let mut qc = QuickCheck {
+            tests: 10000,
+            max_tests: 20000,
+            gen: StdGen::new(rand::thread_rng(), 100, floats_finite_only),
+        };
+
+        qc.quicktest((|n| n.is_finite()) as fn(n: f32) -> bool)
+            .expect("Infinite number was generated for f32");
+
+        qc.quicktest((|n| n.is_finite()) as fn(n: f64) -> bool)
+            .expect("Infinite number was generated for f64");
+    }
+
+    #[test]
+    fn when_floats_finite_only_flag_set_to_false_generates_non_finite_f64() {
+        let floats_finite_only = false;
+
+        let mut gen = StdGen::new(rand::thread_rng(), 100, floats_finite_only);
+        let mut non_finite_was_found = false;
+
+        for _ in 0..10000 {
+            let f = f64::arbitrary(&mut gen);
+            if !f.is_finite() {
+                non_finite_was_found = true;
+                break;
+            }
+        }
+
+        assert!(non_finite_was_found);
+    }
+
+    #[test]
+    fn when_floats_finite_only_flag_set_to_false_generates_non_finite_f32() {
+        let floats_finite_only = false;
+
+        let mut gen = StdGen::new(rand::thread_rng(), 100, floats_finite_only);
+        let mut non_finite_was_found = false;
+
+        for _ in 0..10000 {
+            let f = f32::arbitrary(&mut gen);
+            if !f.is_finite() {
+                non_finite_was_found = true;
```

Comment:
This test can theoretically create a false negative, but probability of that is (as far as I've managed to calculate) 1 of 500.
Maybe number of iterations should be increased, should it?